ddtrace/contrib/internal/langchain/patch.py

```diff
@@ -975,6 +976,165 @@ def traced_similarity_search(langchain, pin, func, instance, args, kwargs):
+                    )
+                elif isinstance(message, langchain_core.prompt_values.PromptValue):
+                    for langchain_message_idx, langchain_message in enumerate(message.messages):
+                        span.set_tag_str(
+                            "langchain.request.messages.%d.%d.content" % (message_idx, langchain_message_idx),
+                            integration.trunc(str(langchain_message.content)),
+                        )
+                        span.set_tag_str(
+                            "langchain.request.messages.%d.%d.role" % (message_idx, langchain_message_idx),
+                            str(langchain_message.__class__.__name__),
+                        )
+                elif isinstance(message, langchain_core.messages.BaseMessage):
+                    span.set_tag_str(
+                        "langchain.request.messages.%d.content" % (message_idx), integration.trunc(str(message.content))
+                    )
+                    span.set_tag_str(
+                        "langchain.request.messages.%d.role" % (message_idx), str(message.__class__.__name__)
+                    )
+                else:
+                    span.set_tag_str(
+                        "langchain.request.messages.%d.content" % (message_idx), integration.trunc(message)
+                    )
+
+        for param, val in getattr(instance, "_identifying_params", {}).items():
+            if isinstance(val, dict):
+                for k, v in val.items():
+                    span.set_tag_str("langchain.request.%s.parameters.%s.%s" % (llm_provider, param, k), str(v))
+            else:
+                span.set_tag_str("langchain.request.%s.parameters.%s" % (llm_provider, param), str(val))
+
+    def _on_span_finished(span: Span, streamed_chunks, error: Optional[bool] = None):
+        if not error and integration.is_pc_sampled_span(span):
+            content = "".join([str(chunk.content) for chunk in streamed_chunks])
+            span.set_tag_str("langchain.response.content", integration.trunc(content))
+
+            usage = getattr(streamed_chunks[-1], "usage_metadata", None)
+            if usage:
+                for k, v in usage.items():
+                    span.set_tag_str("langchain.response.usage_metadata.%s" % k, str(v))
+
+    return shared_stream(
+        integration=integration,
+        pin=pin,
+        func=func,
+        instance=instance,
+        args=args,
+        kwargs=kwargs,
+        interface_type="chat_model",
+        on_span_started=_on_span_started,
+        on_span_finished=_on_span_finished,
+        api_key=_extract_api_key(instance),
+        provider=llm_provider,
+    )
+
+
+@with_traced_module
+def traced_llm_stream(langchain, pin, func, instance, args, kwargs):
+    integration: LangChainIntegration = langchain._datadog_integration
+    llm_provider = instance._llm_type
+
+    def _on_span_start(span: Span):
+        inp = get_argument_value(args, kwargs, 0, "input")
```

Comment:
Does LLM.stream() also accept multiple types of inputs that we need to account for?

Reply:
ah yes good catch, it actually does look like it accepts a union of different values (although, different from the set accepted by chat models). will update to more accurately reflect that!

Reply:
meh, as long as we're not making up roles when they don't exist in `llm.generate()` cases, I'd rather push for a unified way of tagging llms vs chat models, i.e. treat them all as messages. I think this is fine